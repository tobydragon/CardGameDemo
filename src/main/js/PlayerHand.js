import React from "react";
import Row from "react-bootstrap/Row";
import Col from "react-bootstrap/Col";
import Container from "react-bootstrap/Container";
import Button from "react-bootstrap/Button";
import ButtonToolbar from "react-bootstrap/ButtonToolbar";
import Hand from "./Hand";
import Jumbotron from "react-bootstrap/Jumbotron";

export default function PlayerHand(props) {
    return (
        <Container>
            <Row>
                <Col>
                    <Hand ownerName={props.ownerName} cards={props.cards}/>
                    <br/>
                    <HandButtons gameState={props.gameState}
                                 onHitClick={props.handleHit}
                                 onDealClick={props.handleDeal}
                                 onStandClick={props.handleStand}
                    />
                </Col>
            </Row>
        </Container>
    );
}

class HandButtons extends React.Component {
    constructor(props) {
        super(props);

        this.handleHitClick = this.handleHitClick.bind(this);
        this.handleDealClick = this.handleDealClick.bind(this);
        this.handleStandClick= this.handleStandClick.bind(this);

        this.state = {
            onHitClick: props.onHitClick,
            onDealClick: props.onDealClick,
            onStandClick: props.onStandClick,
        }

    }

    handleHitClick(e) {
        this.state.onHitClick();
    }

    handleDealClick(e) {
        this.state.onDealClick();
    }

    handleStandClick(e){
        this.state.onStandClick();
    }

    render() {
        if (this.props.gameState === "PLAYING"){
            console.log("PlayerHand.render: PLAYING");
            return (
                <ButtonToolbar>
                    <Button disabled={true} onClick={this.handleDealClick}>Deal</Button>
                    <Button onClick={this.handleHitClick}>Hit</Button>
                    <Button onClick={this.handleStandClick}>Stand</Button>
                </ButtonToolbar>
            );
        }
        else {
            return (
                <ButtonToolbar>
                    <Button onClick={this.handleDealClick}>Deal</Button>
                    <Button disabled={true} onClick={this.handleHitClick}>Hit</Button>
                    <Button disabled={true} onClick={this.handleStandClick}>Stand</Button>
                </ButtonToolbar>
            );
        }
    }
}

