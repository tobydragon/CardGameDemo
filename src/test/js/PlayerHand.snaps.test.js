import renderer from "react-test-renderer";
import React, { useState } from "react";
import PlayerHand from "../../main/js/PlayerHand";
import {regHandEx} from "./Hand.examples";
import Round from "../../main/js/Round";

// let playerHandResult = shallow(<PlayerHand ownername={"player"} cards={regHandEx}/>);


test('PLAYING state test', () => {
    const component = renderer.create(<PlayerHand gameState={"PLAYING"} ownername={"player"} cards={regHandEx}/>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('not PLAYING state test', () =>{
    const component = renderer.create(<PlayerHand ownername={"player"} cards={[]} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});