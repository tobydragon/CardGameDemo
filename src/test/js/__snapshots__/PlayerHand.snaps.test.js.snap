// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PLAYING state test 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <div
        className="container"
      >
        <div
          className="cardDeck card-group"
        >
          <div
            className="card"
          >
            <div
              className="card-title h5"
            >
              6
            </div>
            <div
              className="card-title h5"
            >
               of
            </div>
            <div
              className="card-title h5"
            >
              diamonds
            </div>
          </div>
          <div
            className="card"
          >
            <div
              className="card-title h5"
            >
              4
            </div>
            <div
              className="card-title h5"
            >
               of
            </div>
            <div
              className="card-title h5"
            >
              hearts
            </div>
          </div>
          <div
            className="card"
          >
            <div
              className="card-title h5"
            >
              8
            </div>
            <div
              className="card-title h5"
            >
               of
            </div>
            <div
              className="card-title h5"
            >
              clubs
            </div>
          </div>
        </div>
      </div>
      <br />
      <div
        className="btn-toolbar"
        role="toolbar"
      >
        <button
          className="btn btn-primary"
          disabled={true}
          onClick={[Function]}
          type="button"
        >
          Deal
        </button>
        <button
          className="btn btn-primary"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Hit
        </button>
        <button
          className="btn btn-primary"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Stand
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`not PLAYING state test 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <div
        className="container"
      >
        <div
          className="cardDeck card-group"
        />
      </div>
      <br />
      <div
        className="btn-toolbar"
        role="toolbar"
      >
        <button
          className="btn btn-primary"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Deal
        </button>
        <button
          className="btn btn-primary"
          disabled={true}
          onClick={[Function]}
          type="button"
        >
          Hit
        </button>
        <button
          className="btn btn-primary"
          disabled={true}
          onClick={[Function]}
          type="button"
        >
          Stand
        </button>
      </div>
    </div>
  </div>
</div>
`;
